### PR TITLE
[arduino-cmake] The ARDUINO_C_FLAGS is set inside a function and the …

### PR DESCRIPTION
**Pull request by mean00**
ARDUINO_C_FLAGS is changed inside a function so the change is lost upon exiting. The problem is that it is reused later on when setting the ARDUINO_CXX_FLAGS. The effect is we lose -mcall-prologues -ffunction-sections -fdata-sections for C++ files,  so the final size is 10% bigger. 
Quick and dirty fix : Mark it as cache internal so the change sticks

### CODE REVIEW

**Review by MrPointer — CHANGES_REQUESTED** [review]
Hmm, this is interesting.
Someone did report a larger program size than expected in #67, however the proposed solution isn't related to the flags at all. 
Without you, probably nobody would've ever noticed it, so you deserve a **big** thank you!

However, before I approve this, I must request you to make one more change: 
Although it doesn't affect program's size, the `setup_cxx_flags` function also has a `set` call which doesn't set it to cache. It looks like this:
```cmake
if (NOT DEFINED ARDUINO_CXX_FLAGS)
        set(ARDUINO_CXX_FLAGS "${ARDUINO_C_FLAGS} -fno-exceptions")
endif (NOT DEFINED ARDUINO_CXX_FLAGS)
```

Sou could you please set it to cache as well? so we could solve it all in a single quite-related PR instead of separating them. Thanks in advance! 😃

**Comment by mean00** [comment]
I added a 2nd commit to address the CXX flags too
Thanks

**Review by MrPointer — DISMISSED** [review]

**Comment by MrPointer** [comment]
I can't merge the PR since the **AppVeyor** build hasn't reported yet.
A look at **AppVeyor**'s build history shows that the build has never even been triggered.
The most frustrating thing is that I can't re-trigger it somehow. @Souler @noseglasses Have any suggestions?

**Comment by noseglasses** [comment]
I am also quite new to using Appveyor. No idea if or how it is possible to manually trigger a PR build. 

As an attempt to solve this problem, I added a tiny space at the end of one of the lines that were changed by this PR. This triggered a new Travis CI build that completed successfully. The Appveyor CI build is, however, still in a weird state where it reports "Waiting for status to be reported ".

@MrPointer: Did other Appveyor builds complete successfully in the meantime?

Edit 1: I can answer my own question as I just submitted another PR. Both, Travis CI and Appveyor completed.

**Comment by MrPointer** [comment]
Well it seems like this PR has never been triggered and **never will**...
As a last resort, I'll open the PR myself with full credits to @mean00 for his contribution.

#### Edit
In fact, when I look at this PR now, I realize that it is actually a good thing - Our [Contribution Guidelines](https://github.com/arduino-cmake/arduino-cmake/blob/master/CONTRIBUTING.md) state that PRs won't be merged to `master` unless it is a vital bug-fix/hotfix.
I will PR it to develop instead.

**Comment by MrPointer** [comment]
Attempting to "re-PR" in #72.

**Pull request closed without merging**
